### PR TITLE
Duplicate checking improvements

### PR DESCRIPTION
**Pull request by andrewconner**
- Slicked
- Persists duplicate documents in db
- Higher efficiency
- Less redundant duplicates stored/sent
- Formatting improvements

### CODE REVIEW

**Comment by andrewconner** [comment]
Implementing separately:
- Cron system for Shoebox
- Process that runs every day to find duplicate documents
- `Unscrapable` model. For now, they're hardcoded (google, facebook, asana). This is temporary.

File: shoebox/app/com/keepit/controllers/admin/ScraperController.scala [hunk 1]
```diff
@@ -32,28 +33,58 @@ object ScraperController extends FortyTwoController {
     Ok(views.html.scrape(articles))
   }
 
+  // These will be removed when the unscrapeable documents are reprocessed
+  val IGNORED_DOCUMENTS = Set(
```
Comment (eishay) [inline shoebox/app/com/keepit/controllers/admin/ScraperController.scala:37]: i thought we'll do it by domain name. lets talk about it, i would like to understand.

File: shoebox/app/com/keepit/model/DuplicateDocument.scala [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    def uri2Id = column[Id[NormalizedURI]]("uri2_id", O.NotNull)
+    def percentMatch = column[Double]("percent_match", O.NotNull)
+    def state = column[State[DuplicateDocument]]("state", O.NotNull)
+    def * = id.? ~ createdAt ~ updatedAt ~ uri1Id ~ uri2Id ~ percentMatch ~ state <> (DuplicateDocument, DuplicateDocument.unapply _)
+  }
+
+  def all(userId: Id[User])(implicit session: RSession): Seq[DuplicateDocument] =
+    (for(f <- table if f.userId === userId && f.state === DuplicateDocumentStates.ACTIVE) yield f).list
+
+  def getSimilarTo(id: Id[NormalizedURI])(implicit session: RSession): Seq[DuplicateDocument] = {
```
Comment (eishay) [inline shoebox/app/com/keepit/model/DuplicateDocument.scala:68]: i would test that, its a bit complex query

File: shoebox/app/com/keepit/controllers/admin/ScraperController.scala (hunk 1 above)
Comment (andrewconner) [inline shoebox/app/com/keepit/controllers/admin/ScraperController.scala:37]: This is temporary. When the `Unscrapable` model is finished, the documents wouldn't get this far. However, for now, there are problematic documents that make the duplicate detection give many, many results. This is just a last-catch filter for those.

File: shoebox/app/com/keepit/model/NormalizedURI.scala [hunk 3]
```diff
@@ -55,6 +60,36 @@ case class NormalizedURI  (
   def stats()(implicit conn: Connection): NormalizedURIStats = NormalizedURIStats(this, BookmarkCxRepo.ofUri(this))
 }
 
+@ImplementedBy(classOf[NormalizedURIRepoImpl])
+trait NormalizedURIRepo extends DbRepo[NormalizedURI]  {
+  def all()(implicit session: RSession): Seq[NormalizedURI]
```
Comment (eishay) [inline shoebox/app/com/keepit/model/NormalizedURI.scala]: maybe change it to all(state) or allActive() ?
you're overriding and changing the `all` method's contract

**Pull request merged**